code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,213 @@
+//Bureaucracy machine!
+//Simply set this up in the hopline and you can serve people based on ticket numbers
+
+/obj/machinery/ticket_machine
+	name = "ticket machine"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticketmachine"
+	desc = "A marvel of bureaucratic engineering encased in an efficient plastic shell. It can be refilled with a hand labeler refill roll and linked to buttons with a multitool."
+	density = FALSE
+	maptext_height = 26
+	maptext_width = 32
+	maptext_x = 7
+	maptext_y = 10
+	layer = HIGH_OBJ_LAYER
+	var/ticket_number = 0 //Increment the ticket number whenever the HOP presses his button
+	var/current_number = 0 //What ticket number are we currently serving?
+	var/max_number = 100 //At this point, you need to refill it.
+	var/cooldown = 50
+	var/ready = TRUE
+	var/list/ticket_holders = list()
+	var/list/tickets = list()
+	var/id = 1
+
+/obj/machinery/ticket_machine/Destroy()
+	if(length(tickets))
+		for(var/obj/item/ticket_machine_ticket/ticket in tickets)
+			ticket.visible_message("<span class='notice'>\the [ticket] disperses!</span>")
+			qdel(ticket)
+		tickets.Cut()
+	ticket_holders.Cut()
+	return ..()
+
+/obj/machinery/ticket_machine/emag_act(mob/user) //Emag the ticket machine to dispense burning tickets, as well as randomize its number to destroy the HoP's mind.
+	if(emagged)
+		return
+	to_chat(user, "<span class='warning'>You overload [src]'s bureaucratic logic circuitry to its MAXIMUM setting.</span>")
+	ticket_number = rand(0, max_number)
+	current_number = ticket_number
+	emagged = TRUE
+	if(length(tickets))
+		for(var/obj/item/ticket_machine_ticket/ticket in tickets)
+			ticket.visible_message("<span class='notice'>\the [ticket] disperses!</span>")
+			qdel(ticket)
+		tickets.Cut()
+	update_icon()
+
+/obj/machinery/ticket_machine/Initialize(mapload)
+	. = ..()
+	update_icon()
+
+/obj/machinery/ticket_machine/proc/increment()
+	if(current_number > ticket_number)
+		return
+	if(current_number && !(emagged) && tickets[current_number])
+		var/obj/item/ticket_machine_ticket/ticket = tickets[current_number]
+		ticket.audible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
+		qdel(tickets[current_number])
+	if(current_number < ticket_number)
+		current_number ++ //Increment the one we're serving.
+		playsound(src, 'sound/misc/announce_dig.ogg', 50, FALSE)
+		atom_say("Now serving ticket #[current_number]!")
+		if(!(emagged) && tickets[current_number])
+			var/obj/item/ticket_machine_ticket/ticket = tickets[current_number]
+			ticket.audible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
+		update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+
+/obj/machinery/ticket_machine_button
+	name = "increment ticket counter"
+	desc = "Use this button after you've served someone to tell the next person to come forward."
+	icon = 'icons/obj/objects.dmi'
```

Comment:
TG uses the big square button.

The icon is under `icon/obj/stationobjs.dmi`
icon_state = "doorctrl0"